data/statuses.js

```diff
@@ -321,7 +321,7 @@ exports.BattleStatuses = {
 		onBeforeMove: function (pokemon) {
 			this.add('cant', pokemon, 'recharge');
 			pokemon.removeVolatile('mustrecharge');
-			return false;
+			return undefined;
```

Comment:
...does this actually work?

Reply:
Are you interpreting `undefined` differently from usual?

Reply:
It works because stomping tantrum exclusively checks for a value of false, so literally anything else doesn’t result in a boost. I could just as easily make this just `return;` or `return “recharge”;` and it would work.

Reply:
I used `undefined` because that’s the only other falsy value `.moveThisTurnSucceeded` ever has stored.

Reply:
No actually it has to be a falsy value so the string option doesn’t work unless that gets hard-coded for.

Reply:
If you don’t like `undefined` and you don’t like `null`, that just leaves 0 or an empty container.

Reply:
No, I mean, in an event handler, returning `undefined` means "do the default thing", which is "don't prevent the move".

Reply:
https://github.com/MacChaeger/Pokemon-Showdown/blob/3051a7c1f49f449d5d546654e84e831b949933b3/sim/battle.js#L865-L871

is the relevant code. `undefined` is definitely intentionally not treated like other falsy values by the event handler.

Reply:
"Must recharge" should maybe be its own pseudo-move... So `this.add('cant', pokemon, 'recharge')` would just be a `Hit` handler.

Reply:
Do you want me to try something, or should I just let you do that and then rebase?

Reply:
What would you be trying? Is there anywhere else you assume `return undefined` interrupts an event? (it doesn't)

Reply:
Oh I forgot you literally told me to use `undefined` instead of `null` for this https://github.com/Zarel/Pokemon-Showdown/pull/4175#discussion_r153219153

Reply:
I would be trying to do the pseudo-move thing, but I'd rather leave that up to you to do so we don't have to go through all this again with however I do that.

Reply:
Well, yes, I said `undefined` means "not success, not failure", but to be clear it also means "continue on with what you were doing before".